### PR TITLE
GH-2094: Allow execution of single task (without dependencies)

### PR DESCRIPTION
**Pull request by patriksvensson**
Related to #2094

### CODE REVIEW

File: src/Cake.Core/ICakeEngine.cs [hunk 1]
```diff
@@ -55,9 +55,9 @@ public interface ICakeEngine
         /// </summary>
         /// <param name="context">The context.</param>
         /// <param name="strategy">The execution strategy.</param>
-        /// <param name="target">The target to run.</param>
+        /// <param name="settings">The execution settings.</param>
         /// <returns>The resulting report.</returns>
-        Task<CakeReport> RunTargetAsync(ICakeContext context, IExecutionStrategy strategy, string target);
+        Task<CakeReport> RunTargetAsync(ICakeContext context, IExecutionStrategy strategy, ExecutionSettings settings);
```
Comment (mholo65) [inline src/Cake.Core/ICakeEngine.cs]: Could we keep the old method and obsolete that? Don’t think much will break due to this, but still.

Comment (patriksvensson) [reply src/Cake.Core/ICakeEngine.cs]: No, sadly not possible since it then would rely on side-effects.

**Comment by patriksvensson** [comment]
PR have been updated and rebased.

**Comment by devlead** [comment]
@patriksvensson your changes have been merged, thanks for your contribution 👍

**Comment by HebaruSan** [comment]
Is it documented somewhere how to use this?

**Pull request merged**